### DISCUSSION TITLE
Calculating battery %

### DISCUSSION THREAD

**Discussion by richardcastle** [WillCodeForCats/solaredge-modbus-multi#237] [Q&A] [answered]
I'm trying to build some automations around the current state of battery charge as a percentage.

As percentage isn't directly surfaced as part of the modbus data, I was looking at whether I could calculate it using other modbus data. 

Initially I had figured that **solaredgemod_b1_available_energy** and **solaredgemod_b1_maximum_energy** would do the trick, but these seem to be measuring different things.

Has anyone found a way to calculate battery %? The SolarEdge API updates far too slowly for it to be useful for me, so I'd like to try and calculate using modbus.

Thanks!

**Comment by mpredfearn** [comment, marked as answer]
![Screenshot_2023-01-30_15-38-26](https://user-images.githubusercontent.com/12936891/215522534-3b64c1b4-e95d-4660-864a-60fd2749fa69.png)

?

**Comment by richardcastle** [comment]
Oh, that's embarrassing. Yes. It's all there. For some reason it wasn't showing up in my auto-populated dashboard entries and didn't think to double check the raw device entities in settings.

@mpredfearn - Thank you!
@WillCodeForCats - Thanks for validating.

**Reply by mpredfearn** [reply]
:laughing: no worries. 
@richardcastle just in case, if you have batteries and want to replicate the "power flow" sensors, you probably will require template sensors for that - see https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Sensors-for-Power-and-Energy. Have fun :-)
